lib/brains/forms.py

```diff
@@ -75,7 +75,7 @@ def clean_plan(self):
         data = self.cleaned_data['plan']
 
         try:
-            obj = SellerProduct.objects.get(external_id=data)
+            obj = SellerProduct.objects.get(public_id=data)
         except ObjectDoesNotExist:
```

Comment:
I just noticed that this could use a logging statement to show the plan ID used to do the lookup when it fails. Otherwise, debugging might be difficult.